editor.planx.uk/src/pages/FlowEditor/components/Sidebar/Publish/PublishDialog.tsx

```diff
@@ -1,389 +1,342 @@
-    useState<string>();
-  const lastPublisher = useStore((state) => state.lastPublisher);
-
-  const _nestedFlowLastPublishedRequest = useAsync(async () => {
-    const user = await lastPublisher(flowId);
-    setNestedFlowLastPublishedTitle(
-      formatLastPublishMessage(publishedAt, user),
-    );
-  });
+export const ChangesDialog = (props: ChangesDialogProps) => {
+  const {
+    dialogOpen,
+    setDialogOpen,
+    alteredNodes,
+    history,
+    lastPublishedTitle,
+    validationChecks,
+    previewURL,
+    handlePublish,
+  } = props;
 
-  return (
-    <ListItem
-      key={publishedFlowId}
-      disablePadding
-      sx={{ display: "list-item" }}
-    >
-      <ListItemText
-        primary={
-          useStore.getState().canUserEditTeam(text.split("/")[0]) ? (
-            <Link href={`../${text}`} target="_blank">
-              <Typography variant="body2">{text}</Typography>
-            </Link>
-          ) : (
-            <Typography variant="body2">{text}</Typography>
-          )
-        }
-        secondary={
-          <>
-            <Typography variant="body2" fontSize="small">
-              {nestedFlowLastPublishedTitle}
-            </Typography>
-            {summary && (
-              <Typography variant="body2" fontSize="small">
-                {summary}
-              </Typography>
-            )}
-          </>
-        }
-      />
-    </ListItem>
-  );
-};
+  const steps = ["Review", "Test", "Publish"];
+  const [activeStep, setActiveStep] = useState<number>(0);
 
-interface AlteredNodesSummary {
-  title: string;
-  portals: Portal[];
-  updated: number;
-  deleted: number;
-}
-
-export const AlteredNodesSummaryContent = (props: {
-  alteredNodes: AlteredNode[];
-  lastPublishedTitle: string;
-}) => {
-  const { alteredNodes, lastPublishedTitle } = props;
-  const [expandNodes, setExpandNodes] = useState<boolean>(false);
-
-  const changeSummary: AlteredNodesSummary = {
-    title: lastPublishedTitle,
-    portals: [],
-    updated: 0,
-    deleted: 0,
+  const handleNext = () => {
+    setActiveStep(activeStep + 1);
   };
 
-  alteredNodes.map((node) => {
-    if (node.id === "0") {
-      changeSummary["title"] =
-        "You are publishing the main service for the first time.";
-    } else if (node.id && Object.keys(node).length === 1) {
-      changeSummary["deleted"] += 1;
-    } else if (node.type === TYPES.InternalPortal) {
-      if (node.data?.text?.includes("/") && !node.data?.text?.includes(" ")) {
-        changeSummary["portals"].push({ ...node.data, flowId: node.id });
-      }
-    } else if (node.type) {
-      changeSummary["updated"] += 1;
-    }
+  const handleBack = () => {
+    setActiveStep(activeStep - 1);
+  };
 
-    return changeSummary;
-  });
+  const ReviewStep = () => {
+    const [showError, setShowError] = useState<boolean>(false);
+    const atLeastOneFail =
+      validationChecks.filter((check) => check.status === "Fail").length > 0;
```

Comment:
```suggestion
    const atLeastOneFail = validationChecks.some((check) => check.status === "Fail")
```

nit: Maybe a bit more idiomatic / readable?